extensions/amp-story/1.0/amp-story-share-menu.js

```diff
@@ -105,8 +113,10 @@ export class ShareMenu {
 
     this.isBuilt_ = true;
 
+    this.ampdoc_ = getAmpdoc(this.parentEl_);
```

Comment:
Nit: I think it'd be better to keep doing `getAmpDoc()` from `buildForSystemSharing_()`, we never know how the code will evolve and it's hard to rely on the fact that a method runs before another one.
Line `141` is already calling `getAmpDoc()` so I think we should just re-use this.

Reply:
Done.

What I am trying to do here is to avoid paying the cost of getting ampdoc twice. Since we already get one here in build() we should just keep it. But if the cost of it is basically none we can just get it where it is closest to the place it gets used. I guess the cost of getting ampdoc is negligible?